docs/how-to-write-function.md

```diff
@@ -0,0 +1,89 @@
+# A Guide on How to Use Function for Scalar DL
+
+This document sets out some guidelines for writing functions for Scalar DL.
+
+## What is a function for Scalar DL ?
+
+A Function (Smart Function) for Scalar DL is a Java program extending [`Function`](https://scalar-labs.github.io/scalardl/javadoc/ledger/com/scalar/dl/ledger/function/Function.html) class written for implementing a single function of business logic, which mainly manages the data of an application that is deployed on top of Scalar DL.
+Before taking a look at this, it is recommended to check [Getting Started in Scalar DL](dl-getting-started.md) and [Scalar DL v1 design document](dl-design.md) to understand what Scalar DL is and its basic terminologies. Understanding [contact guide](how-to-write-contract.md) is also helpful to undertand this guide properly.
+
+## Background
+
+Assets managed by Contracts in Scalar DL are, as most of the DLT platforms including Blockchains, tamper-evident and append-only so their data structure is limited in modeling wide variety of applications and they are not able to be deleted.
+Many of the DLT platforms deal with the issue by having another database such as a RDBMS in front of a ledger to handle application's data in the database and write logs to the ledger as evidence.
+However this scheme is not ideal since data consistency in between the database and the ledger is not always preserved. There could be a case where applications don't have corresponding logs in the ledger due to a catastrophic failure, which defeats the purpose of writing logs to the ledger as evidence.
```

Comment:
```suggestion
However, this scheme is not ideal since data consistency between the database and the ledger is not always preserved. There could be a case where applications don't have corresponding logs in the ledger due to a catastrophic failure, which defeats the purpose of writing logs to the ledger as evidence.
```